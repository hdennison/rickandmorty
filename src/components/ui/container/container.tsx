import { Slot } from "@radix-ui/react-slot";
import { cva, type VariantProps } from "class-variance-authority";
import { cn } from "../utils";
import { ReactNode } from "react";

const panelVariants = cva("mx-auto w-full px-4 md:px-6 xl:px-0", {
  variants: {
    width: {
      lg: "max-w-screen-lg",
    },
  },
  defaultVariants: {
    width: "lg",
  },
});

export interface ContainerProps
  extends React.HTMLAttributes<HTMLDivElement>,
    VariantProps<typeof panelVariants> {
  asChild?: boolean;
  children: ReactNode;
}

export const Container = ({
  className,
  children,
  width,
  asChild = false,
  ...props
}: ContainerProps) => {
  const Comp = asChild ? Slot : "div";

  return (
    <Comp className={cn(panelVariants({ width, className }))} {...props}>
      {children}
    </Comp>
  );
};
