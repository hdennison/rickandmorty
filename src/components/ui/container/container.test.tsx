import { expect, test } from "vitest";
import { render } from "vitest-browser-react";
import { Container } from "./container";

test("Renders as wrapper", () => {
  const container = render(<Container id="container">Foo</Container>);

  // Snapshots are not the best solution, it woould be much better to set up Storybook + Chromatic
  expect(container).toMatchSnapshot();
});

test("Renders asChild", () => {
  const container = render(
    <Container id="container" asChild>
      <article>Bar</article>
    </Container>,
  );

  // Snapshots are not the best solution, it woould be much better to set up Storybook + Chromatic
  expect(container).toMatchSnapshot();
});
