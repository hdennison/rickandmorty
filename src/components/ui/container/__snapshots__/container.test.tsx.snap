// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Renders as wrapper 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    
    
    
  


    <div>
      <div
        class="mx-auto w-full max-w-screen-lg"
        id="container"
      >
        Foo
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mx-auto w-full max-w-screen-lg"
      id="container"
    >
      Foo
    </div>
  </div>,
  "debug": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholder": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Renders asChild 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    
    
    
  


    <div>
      <article
        class="mx-auto w-full max-w-screen-lg"
        id="container"
      >
        Bar
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="mx-auto w-full max-w-screen-lg"
      id="container"
    >
      Bar
    </article>
  </div>,
  "debug": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholder": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
